src/components/table/JadeTable.jsx

```diff
@@ -1,207 +1,171 @@
-    marginTop: theme.spacing(3),
-    marginRight: theme.spacing(2),
-    marginLeft: 0,
-  },
-  searchIcon: {
-    color: theme.palette.primary.main,
-    width: 22,
-    height: '100%',
-    position: 'absolute',
-    pointerEvents: 'none',
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-    paddingLeft: theme.spacing(3),
   },
-  searchInput: {
-    paddingTop: theme.spacing(1.5),
-    paddingLeft: theme.spacing(6),
+  tableWrapper: {
+    maxHeight: 'calc(100vh - 300px)',
+    overflow: 'auto',
   },
 });
 
-const DEFAULT_PAGE_SIZE = 10;
-
 export class JadeTable extends React.PureComponent {
-  state = {
-    orderDirection: 'desc',
-    orderBy: 'created_date',
-    page: 0,
-    rowsPerPage: DEFAULT_PAGE_SIZE,
-    searchString: '',
-  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      page: 0,
+      rowsPerPage: 100,
+      pageToTokenMap: {},
+      orderBy: '',
+      order: '',
+    };
+  }
 
   static propTypes = {
-    classes: PropTypes.object.isRequired,
-    columns: PropTypes.arrayOf(PropTypes.object),
-    handleEnumeration: PropTypes.func,
-    itemType: PropTypes.string.isRequired,
-    rowKey: PropTypes.func,
-    rows: PropTypes.arrayOf(PropTypes.object),
-    summary: PropTypes.bool,
-    totalCount: PropTypes.number,
+    classes: PropTypes.object,
+    columns: PropTypes.array,
+    dispatch: PropTypes.func.isRequired,
+    queryResults: PropTypes.object,
+    rows: PropTypes.array,
   };
 
-  handleRequestSort = (event, sort) => {
-    const { handleEnumeration } = this.props;
-    const { orderDirection, orderBy, page, rowsPerPage, searchString } = this.state;
-    let newOrder = 'desc';
-    if (orderBy === sort && orderDirection === 'desc') {
-      newOrder = 'asc';
+  handleChangePage = (event, newPage) => {
+    const { dispatch, queryResults } = this.props;
+    const { page, rowsPerPage, pageToTokenMap } = this.state;
+
+    if (page === 0) {
+      pageToTokenMap[0] = undefined;
+    }
+
+    if (newPage > page) {
+      pageToTokenMap[newPage] = queryResults.pageToken;
     }
-    const offset = page * rowsPerPage;
-    this.setState({ orderDirection: newOrder, orderBy: sort });
-    handleEnumeration(rowsPerPage, offset, sort, newOrder, searchString);
+
+    dispatch(
+      pageQuery(
+        pageToTokenMap[newPage],
+        queryResults.jobReference.projectId,
+        queryResults.jobReference.jobId,
+        rowsPerPage,
+      ),
+    );
+    this.setState({
+      page: newPage,
+      pageToTokenMap,
+    });
   };
 
   handleChangeRowsPerPage = event => {
-    const { handleEnumeration } = this.props;
-    const { orderDirection, orderBy, page, searchString } = this.state;
-    const limit = event.target.value;
-    const offset = page * limit;
-    this.setState({ rowsPerPage: limit });
-    handleEnumeration(limit, offset, orderBy, orderDirection, searchString);
+    this.setState({
+      rowsPerPage: +event.target.value,
```

Comment:
I used to do always do this, but found it's confusing to readers unfamiliar with the quirks of js. Better to do the less-cool-but-more-readable `parseInt`